### PR TITLE
Cypher builder string functions

### PR DESCRIPTION
**Pull request by angrykoala**
# Description
Exposes the `Function` class for arbitrary functions and add the string functions

### CODE REVIEW

File: packages/cypher-builder/src/Cypher.ts [hunk 1]
```diff
@@ -93,6 +95,9 @@ export {
     sum,
     randomUUID,
 } from "./expressions/functions/CypherFunction";
+
+export * from "./expressions/functions/StringFunctions";
```
Comment (MacondoExpress) [inline packages/cypher-builder/src/Cypher.ts:99]: Good that we start to categorize them!

**Review by MacondoExpress — COMMENTED** [review]
I think you forgot to include the CypherFunction class in the commit!

**Comment by angrykoala** [comment]
I'm not sure what do you mean. The Function class is now exported normally, instead of as a type

File: packages/cypher-builder/src/expressions/functions/StringFunctions.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+export function rTrim(original: Expr): CypherFunction {
+    return new CypherFunction("rTrim", [original]);
+}
+
+export function split(original: Expr, delimiter: Expr): CypherFunction {
+    return new CypherFunction("split", [original, delimiter]);
+}
+
+export function substring(original: Expr, start: Expr, length?: Expr): CypherFunction {
+    return new CypherFunction("substring", filterTruthy([original, start, length]));
```
Comment (darrellwarde) [inline packages/cypher-builder/src/expressions/functions/StringFunctions.ts:54]: Why the `filterTruthy` here? Could result in some unexpected behaviour no?

**Review by darrellwarde — APPROVED** [review]
LGTM! Just one question.

File: packages/cypher-builder/src/expressions/functions/StringFunctions.ts (hunk 2 above)
Comment (angrykoala) [reply packages/cypher-builder/src/expressions/functions/StringFunctions.ts:54]: The main reason is the `length` argument being optional. At the moment, CypherFunction class do not accept undefined parameters.

The only unexpected behaviour would be if a JavaScript user passes undefined in the first 2 parameters. But that would generate errors anyway

**Comment by MacondoExpress** [comment]
Don't worry I forgot that was actually the current name of the Function class, so I thought it was a newer class 🤦

**Pull request merged**